src/components/ec-modal/ec-modal.vue

```diff
@@ -99,132 +99,123 @@
-        this.$emit('update:show', false);
-        this.$emit('close', false);
-      }
-    },
-    escapeIsPressed(e) {
-      if (e.keyCode === KeyCode.ESC) {
-        this.closeModal();
-      }
-    },
-    getFocusTrapOptions() {
-      const options = {
-        escapeDeactivates: this.isClosable,
-        clickOutsideDeactivates: this.isClosable,
-      };
-
-      if (this.hasPositiveButton()) {
-        options.initialFocus = /* istanbul ignore next */ () => this.$refs.positiveButton.$el;
-      } else if (this.hasNegativeButton()) {
-        options.initialFocus = /* istanbul ignore next */ () => this.$refs.negativeButton.$el;
-      }
+});
 
-      return options;
-    },
-    hasPositiveButton() {
-      return !!this.$slots.positive;
-    },
-    hasNegativeButton() {
-      return !!this.$slots.negative;
-    },
-    hasFooterLeftContent() {
-      return !!this.$slots.footerLeftContent;
-    },
-    hasFooter() {
-      return this.hasFooterLeftContent() || this.hasPositiveButton() || this.hasNegativeButton();
-    },
-  },
-};
+const emit = defineEmits(['update:show', 'negative', 'positive', 'close']);
+
+const positiveButton = ref(null);
+const negativeButton = ref(null);
+
+const isLoadingPositiveButton = computed(() => !!props.isLoading.positive);
+const isLoadingNegativeButton = computed(() => !!props.isLoading.negative);
+const positiveButtonCategory = computed(() => props.category.positive || 'primary');
+const negativeButtonCategory = computed(() => props.category.negative || 'secondary');
+const zIndexStyle = computed(() => (props.zIndex ? { zIndex: props.zIndex } : null));
+
+onBeforeUnmount(() => {
+  document.removeEventListener('keyup', escapeIsPressed);
+});
+
+function negativeAction() {
+  emit('negative');
+}
+
+function positiveAction() {
+  emit('positive');
+}
+
+function closeModal() {
+  if (props.isClosable) {
+    emit('update:show', false);
+    emit('close', false);
+  }
+}
+
+function escapeIsPressed(e) {
+  if (e.keyCode === KeyCode.ESC) {
+    closeModal();
+  }
+}
+
+function getFocusTrapOptions() {
+  const options = {
+    escapeDeactivates: props.isClosable,
+    clickOutsideDeactivates: props.isClosable,
+  };
+
+  if (hasPositiveButton()) {
+    options.initialFocus = /* istanbul ignore next */ () => positiveButton.value.$el;
+  } else if (hasNegativeButton()) {
+    options.initialFocus = /* istanbul ignore next */ () => negativeButton.value.$el;
+  }
+
+  return options;
+}
+
+function hasPositiveButton() {
+  return !!slots.positive;
+}
+
+function hasNegativeButton() {
+  return !!slots.negative;
+}
+
+function hasFooterLeftContent() {
+  return !!slots.footerLeftContent;
+}
+
+function hasFooter() {
+  return hasFooterLeftContent() || hasPositiveButton() || hasNegativeButton();
+}
+
+watchEffect(() => {
+  const value = props.show;
+  if (value) {
+    document.addEventListener('keyup', escapeIsPressed);
+  } else {
+    document.removeEventListener('keyup', escapeIsPressed);
+  }
+});
```

Comment:
Similar thing as in phone-number-input, we should group logic together, now when we don't have to follow a structure of options API.

```
// keyboard nav
watchEffect(() => {
onBeforeUnmount(() => {
  document.removeEventListener('keyup', escapeIsPressed);
});
function escapeIsPressed(e) {
function closeModal() {

// buttons
const positiveButton
const negativeButton
const isLoadingPositiveButton
const isLoadingNegativeButton
const positiveButtonCategory
const negativeButtonCategory
function hasPositiveButton() {
function hasNegativeButton() {

// focus trap
getFocusTrapOptions

// footer
function hasFooterLeftContent() {
function hasFooter() {

// z-index
const zIndexStyle 
```